controllers/cks_cluster_controller.go

```
@@ -0,0 +1,124 @@
+func NewCksClusterReconciliationRunner() *CksClusterReconciliationRunner {
+	// Set concrete type and init pointers.
+	runner := &CksClusterReconciliationRunner{ReconciliationSubject: &infrav1.CloudStackCluster{}}
+	runner.FailureDomains = &infrav1.CloudStackFailureDomainList{}
+	// Setup the base runner. Initializes pointers and links reconciliation methods.
+	runner.ReconciliationRunner = csCtrlrUtils.NewRunner(runner, runner.ReconciliationSubject, "CKSClusterController")
+	runner.CSCluster = runner.ReconciliationSubject
+	return runner
+}
+
+// Reconcile is the method k8s will call upon a reconciliation request.
+func (reconciler *CksClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (retRes ctrl.Result, retErr error) {
+	r := NewCksClusterReconciliationRunner()
+	r.UsingBaseReconciler(reconciler.ReconcilerBase).ForRequest(req).WithRequestCtx(ctx)
+	r.WithAdditionalCommonStages(r.GetFailureDomains(r.FailureDomains))
+	return r.RunBaseReconciliationStages()
+}
+
+// Reconcile actually reconciles the CloudStackCluster.
+func (r *CksClusterReconciliationRunner) Reconcile() (res ctrl.Result, reterr error) {
+	if r.CSCluster.Spec.SyncWithACS {
+		// Prevent premature deletion.
+		controllerutil.AddFinalizer(r.ReconciliationSubject, CksClusterFinalizer)
+
+		if len(r.FailureDomains.Items) == 0 {
+			return r.RequeueWithMessage("No failure domains found")
+		}
+
+		res, err := r.AsFailureDomainUser(&r.FailureDomains.Items[0].Spec)()
+		if r.ShouldReturn(res, err) {
+			return res, err
+		}
+
+		r.Log.Info("Creating entry with CKS")
+		err = r.CSUser.GetOrCreateCksCluster(r.CAPICluster, r.ReconciliationSubject, &r.FailureDomains.Items[0].Spec)
+		if err != nil {
+			if strings.Contains(err.Error(), "Kubernetes Service plugin is disabled") {
+				r.Log.Info("Kubernetes Service plugin is disabled on CloudStack. Skipping ExternalManaged kubernetes cluster creation")
+				return ctrl.Result{}, nil
```

should we just return an error? if the plugin is not enabled in CS but the cluster is configured to use CKS cluster, this is a configuration error, right? or at least log it as an error